text/3550-new-range.md

```diff
@@ -0,0 +1,441 @@
+
+For example: while experimenting with implementing the library part of these changes, we had to modify [`quote`](https://github.com/pitaj/quote/commit/44feebf0594b255a511ff20890a7acbf4d6aeed1) and [`rustc-rayon`](https://github.com/pitaj/rustc-rayon/commit/e76e554512ce25abb48f4118576ede5d7a457918).
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+## Just implement `Copy` on the types as-is
+
+`Copy` iterators are a large footgun. It was decided to [remove `Copy` from all iterators back in 2015](https://github.com/rust-lang/rust/pull/21809), and that decision is unlikely to be reversed.
+
+Another spin on this is to specialize `IntoIterator` on these range types and lint whenever the `Iterator` impl is used. However, that would likely be a breaking change and this kind of specialization may never be stabilized (and might not even be sound).
+
+Neither of these approaches would resolve the issue of `RangeInclusive` being larger than necessary for range purposes.
+
+## Name the new types something besides `Range`
+
+We could choose to introduce these new types with a name other than `Range`. Some alternatives that have been proposed:
+- Interval
+- Span
+- Bounds
+
+We believe that it is best to keep the `Range` naming for several reasons:
+- Existing `Range*` types that implement `Copy` and not `Iterator` that won't be touched by this change
+- Large amount of legacy educational material and code using the `Range` naming
+- It's best to match the name of the syntax (["range expressions"](https://doc.rust-lang.org/reference/expressions/range-expr.html))
+
+## Use legacy range types as the iterators for the new range types
+
+We could choose to make `new_range.into_iter()` resolve to a legacy range type. This would reduce the number of new types we need to add to the standard library.
+
+But the legacy range types have a much larger API surface than other `Iterator`s in the standard library, which typically only implement the various iterator traits and maybe have a `remainder` method. Using the legacy range types this way would break convention and could reduce optimization potential for the iterators, possibly limiting performance.
+
+# Prior art
+[prior-art]: #prior-art
+
+The [copy-range](https://docs.rs/copy-range) crate provides types similar to those proposed here.
+
+# Unresolved questions
+[unresolved-questions]: #unresolved-questions
+
+The set of inherent methods copied from `Iterator` present on the new range types is left open for the **libs-api** team to decide after this proposal is accepted and before stabilization.
+
+- Should other range-related items (like `RangeBounds`) also be moved under `std::ops::range`?
+- Should `RangeFrom` even implement `IntoIterator`, or should it require an explicit `.iter()` call? Using it as an iterator [can be a footgun](https://github.com/rust-lang/libs-team/issues/304), usually people want `start..=MAX` instead.
```

Comment:
Is it correct that `RangeTo` does not implement `IntoIterator`? That seems like useful information to mention here.

It seems odd indeed that these two types would be inconsistent.